cmd/containerd-shim/shim_linux.go

```diff
@@ -32,64 +25,6 @@ func setupSignals() (chan os.Signal, error) {
 	return signals, nil
 }
 
-func newServer() *grpc.Server {
-	return grpc.NewServer(grpc.Creds(NewUnixSocketCredentials(0, 0)))
-}
-
-type unixSocketCredentials struct {
-	uid        int
-	gid        int
-	serverName string
-}
-
-// NewUnixSocketCredentials returns TransportCredentials for a local unix socket
-func NewUnixSocketCredentials(uid, gid int) credentials.TransportCredentials {
-	return &unixSocketCredentials{uid, gid, "locahost"}
-}
-
-func (u *unixSocketCredentials) ClientHandshake(ctx context.Context, addr string, rawConn net.Conn) (net.Conn, credentials.AuthInfo, error) {
-	return nil, nil, errors.New("ClientHandshake is not supported by unixSocketCredentials")
-}
-
-func (u *unixSocketCredentials) ServerHandshake(c net.Conn) (net.Conn, credentials.AuthInfo, error) {
-	uc, ok := c.(*net.UnixConn)
-	if !ok {
-		return nil, nil, errors.New("unixSocketCredentials only supports unix socket")
-	}
-
-	f, err := uc.File()
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "unixSocketCredentials: failed to retrieve connection underlying fd")
-	}
-	pcred, err := unix.GetsockoptUcred(int(f.Fd()), unix.SOL_SOCKET, unix.SO_PEERCRED)
```

Comment:
@crosbymichael @mlaventure Should we implement this for the new shim protocol?

Reply:
I'd say yes, without this, an abstract socket can be accessed by any user